jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/parser/ProjectResolver.java

```diff
@@ -83,6 +92,210 @@ public static List<DependencyInfo> resolveProjectDependencies(String projectUri,
+     * This handles multi-module Maven/Gradle projects where the parent is not a Java project itself.
+     * Returns aggregated information useful for AI context (Java version, common dependencies, build tool).
+     * 
+     * @param root The workspace root
+     * @param parentPath The path of the parent/aggregator project
+     * @param monitor Progress monitor
+     * @return Aggregated dependency information from all sub-projects
+     */
+    private static List<DependencyInfo> resolveAggregatorProjectDependencies(
+            IWorkspaceRoot root, IPath parentPath, IProgressMonitor monitor) {
+        
+        List<DependencyInfo> result = new ArrayList<>();
+        List<IJavaProject> javaProjects = new ArrayList<>();
+        
+        // Find all Java projects under the parent path
+        IProject[] allProjects = root.getProjects();
+        for (IProject p : allProjects) {
+            if (p.getLocation() != null && parentPath.isPrefixOf(p.getLocation())) {
+                try {
+                    if (p.isAccessible() && p.hasNature(JavaCore.NATURE_ID)) {
+                        IJavaProject jp = JavaCore.create(p);
+                        if (jp != null && jp.exists()) {
+                            javaProjects.add(jp);
+                        }
+                    }
+                } catch (CoreException e) {
+                    // Skip this project
+                }
+            }
+        }
+        
+        if (javaProjects.isEmpty()) {
+            JdtlsExtActivator.logInfo("No Java sub-projects found under: " + parentPath.toOSString());
+            return result;
+        }
+        
+        JdtlsExtActivator.logInfo("Found " + javaProjects.size() + 
+            " Java sub-project(s) under: " + parentPath.toOSString());
+        
+        // Mark as aggregator project
+        result.add(new DependencyInfo("aggregatorProject", "true"));
+        result.add(new DependencyInfo("totalSubProjects", String.valueOf(javaProjects.size())));
+        
+        // Collect sub-project names for reference
+        StringBuilder projectNames = new StringBuilder();
+        for (int i = 0; i < javaProjects.size(); i++) {
+            if (i > 0) projectNames.append(", ");
+            projectNames.append(javaProjects.get(i).getProject().getName());
+        }
+        result.add(new DependencyInfo("subProjectNames", projectNames.toString()));
+        
+        // Determine the primary/representative Java version (most common or highest)
+        String primaryJavaVersion = determinePrimaryJavaVersion(javaProjects);
+        if (primaryJavaVersion != null) {
+            result.add(new DependencyInfo(KEY_JAVA_VERSION, primaryJavaVersion));
+        }
+        
+        // Collect all unique libraries across sub-projects (top 10 most common)
+        Map<String, Integer> libraryFrequency = collectLibraryFrequency(javaProjects, monitor);
```

Comment:
quite complicated logic, can you explain a little bit? And check if it's really necessary.

Reply:
sure, current remove such logic and just update the PR format the return json

Reply:
removed